File: examples/sbt-basic/src/test/scala/example/StringFormatterTest.scala
package example

import org.scalatest.funsuite.AnyFunSuite
import org.scalatest.matchers.should.Matchers

class StringFormatterTest extends AnyFunSuite with Matchers {
  val fmt = new StringFormatter("[", "]")

  test("format should wrap value with prefix and suffix") {
    fmt.format("hello") shouldBe "[hello]"
  }

  test("padLeft should pad shorter strings") {
    fmt.padLeft("42", 5) shouldBe "   42"
  }

  test("padLeft should not pad strings at width") {
    fmt.padLeft("hello", 5) shouldBe "hello"
  }

  test("padRight should pad shorter strings") {
    fmt.padRight("42", 5) shouldBe "42   "
  }

  test("truncate should shorten long strings") {
    fmt.truncate("abcdefgh", 5) shouldBe "abcde"
  }

  test("truncate should leave short strings unchanged") {
    fmt.truncate("abc", 5) shouldBe "abc"
  }

  test("repeat should duplicate string") {
    fmt.repeat("ab", 3) shouldBe "ababab"
  }
}

File: examples/maven-basic/src/test/java/example/StringFormatterTest.java
package example;

import org.junit.jupiter.api.Test;

class StringFormatterTest {

    @Test
    void formatWrapsValueWithPrefixAndSuffix() {
        StringFormatter fmt = new StringFormatter("[", "]");
        String result = fmt.format("hello");

        if (!"[hello]".equals(result)) {
            throw new AssertionError("Expected [hello] but got " + result);
        }
    }

    @Test
    void padLeftPadsShorterStrings() {
        StringFormatter fmt = new StringFormatter("", "");
        String result = fmt.padLeft("42", 5, ' ');

        if (!"   42".equals(result)) {
            throw new AssertionError("Expected '   42' but got '" + result + "'");
        }
    }

    @Test
    void padRightPadsShorterStrings() {
        StringFormatter fmt = new StringFormatter("", "");
        String result = fmt.padRight("42", 5, ' ');

        if (!"42   ".equals(result)) {
            throw new AssertionError("Expected '42   ' but got '" + result + "'");
        }
    }

    @Test
    void truncateShortsLongStrings() {
        StringFormatter fmt = new StringFormatter("", "");
        String result = fmt.truncate("abcdefgh", 5);

        if (!"abcde".equals(result)) {
            throw new AssertionError("Expected 'abcde' but got '" + result + "'");
        }
    }

    @Test
    void repeatDuplicatesString() {
        StringFormatter fmt = new StringFormatter("", "");
        String result = fmt.repeat("ab", 3);

        if (!"ababab".equals(result)) {
            throw new AssertionError("Expected 'ababab' but got '" + result + "'");
        }
    }
}
